### PR TITLE
support min and max values for number field

### PR DESCRIPTION
**Pull request by iMokhles**
add support for minimum and maximum values, also add support for maxLength

### CODE REVIEW

File: src/resources/views/fields/number.blade.php [hunk 1]
```diff
@@ -10,6 +10,10 @@
         	name="{{ $field['name'] }}"
             id="{{ $field['name'] }}"
             value="{{ old($field['name']) ? old($field['name']) : (isset($field['value']) ? $field['value'] : (isset($field['default']) ? $field['default'] : '' )) }}"
+            min="{{ $field['min'] }}"
```
Comment (lloy0076) [inline src/resources/views/fields/number.blade.php]: What happens if they're not set?

Comment (iMokhles) [reply src/resources/views/fields/number.blade.php]: I though about it but hadn't time to fix it for the last 2 days, btw thanks for the review and I will fix that

**Comment by tabacitu** [comment]
Hi @iMokhles ,

Thanks for the PR. 

I have to be honest - I'm not 100% sure we should implement it this way. As far as I know, we can already do that using the "attributes" array, which I personally already use for "steps" for floats. Wouldn't adding support for other attributes, at a lower level, make this a bit confusing? Possible confusing questions:
- Why would I define the min on level 0 and steps on level 1? 
- What if I define the min and max both at level 0 and at level 1, which one will be used?

```php
[   // Number
    'name' => 'number',
    'label' => 'Number',
    'type' => 'number',
    // optionals
    'attributes' => [ // these attributes and their values will be passes to the HTML input
           "step" => "any",
           "min" => 1,
           "max" => 25,
    ], // allow decimals
    // 'prefix' => "$",
    // 'suffix' => ".00",
]
```

I think, since ```attributes``` is a feature of ALL fields (not just number), we should make better use of it. What do you think?

Cheers!

**Comment by iMokhles** [comment]
yes I used to do it with `attributes` so we can keep it as it's with `attributes` and add only `maxlength` and `keyPress` to limit length of input numbers

**Comment by tabacitu** [comment]
Great! Please help me understand ```maxlength``` too. 

Isn't an alternative to ```maxlength = 4``` just specifying ```max = 9999```?

**Comment by iMokhles** [comment]
yes we can use max = 9999, but if the user press 123456 to infinity it will work unless the form stop it after submitting, maxlength + keyPress prevent any input greater than the specified length so we block any input greater than 4 or 5 numbers, field won't accept any input

**Comment by tabacitu** [comment]
Oh ok, I understand. That's a bit atypical, since no other CRUD fields include this kind of enforcing. 

I think it's a good feature for the next version of Backpack\CRUD, when validation will happen as-you-type, but right now I think it's reasonable for this to also be done with ```attributes```, as needed: ```"onKeyPress" => "if(this.value.length === 4) return false;"``` is a pretty easy statement.

So I'll close this PR for now, but I made a note for it for the next version. 
Thanks for using and improving Backpack. Cheers!

**Pull request closed without merging**